test_programs/plonky2_trace/2_function_calls/expected_trace.json

```diff
@@ -0,0 +1,47 @@
+[
+  {
+    "Path": ""
+  },
+  {
+    "Function": {
+      "path_id": 0,
+      "line": -1,
+      "name": "<toplevel>"
+    }
+  },
+  {
+    "Call": {
+      "function_id": 0,
+      "args": []
+    }
+  },
+  {
+    "Type": {
+      "kind": 30,
+      "lang_type": "None",
+      "specific_info": {
+        "kind": "None"
+      }
+    }
+  },
+  {
+    "Path": "?"
+  },
+  { "Step": { "path_id": 1, "line": -1 } },
+  {
+    "Path": "/home/stan/code/repos/noir/test_programs/plonky2_trace/2_function_calls/src/main.nr"
+  },
+  { "Step": { "path_id": 2, "line": 9 } },
+  { "Step": { "path_id": 2, "line": 5 } },
+  { "Step": { "path_id": 2, "line": 1 } },
+  { "Step": { "path_id": 2, "line": 2 } },
+  { "Step": { "path_id": 2, "line": 3 } },
+  { "Step": { "path_id": 2, "line": 7 } },
+  { "Step": { "path_id": 2, "line": 5 } },
+  { "Step": { "path_id": 2, "line": 1 } },
+  { "Step": { "path_id": 2, "line": 2 } },
+  { "Step": { "path_id": 2, "line": 3 } },
+  { "Step": { "path_id": 2, "line": 7 } },
+  { "Step": { "path_id": 2, "line": 12 } },
+  { "Step": { "path_id": 2, "line": 13 } }
```

Comment:
Could map this with the logic in `test_programs/plonky2_trace/2_function_calls/src/main.nr` Looks nice.

But what happens when you encounter ` assert(y != 0);`? Maybe would be nice to direct you to the implementation of the `assert`?

Reply:
assert is special here; think of it as a keyword, not a function call. In other words: there is no implementation.

Reply:
Good!
But if it was? My point is, what would happen if we call a function coming from another file?

Reply:
Good Q. Also merits a test. I expect that It Just Works TM, meaning the new file will be cited in the trace and the correct lines in it will be shown.